api/controllers/expense.controller.js

```diff
@@ -1,5 +1,80 @@
+const moment = require('moment');
+
+const Wallet = require('../models/wallet.model');
+const { CREATED_STATUS } = require('../constants/httpStatus.constant');
+
 module.exports.addExpense = async (req, res) => {
-	// const date = new Date(req.body.date);
+	const { date, expense, isIncome, title, description, walletName } = req.body;
+	const { _id } = req.user;
+	const now = moment(date).format('MMMM Do YYYY');
+
+	// Find transactions with the same date
+	const wallet = await Wallet.findOne({
+		$and: [{ walletName }, { owner: _id }],
+	});
+
+	const transactionIndex = wallet.transactions.findIndex(
+		(obj) => moment(obj.date).format('MMMM Do YYYY') === now
+	);
+
+	const currentAccountBalance = wallet.accountBalance;
+	const newExpense = (isIncome && expense) || -expense;
```

Comment:
```suggestion
	const newExpense = isIncome ? expense : -expense;
```
cu Hiếu thử cái ni thử, anh thử thấy được, như cái cũ là isIncome true hay false chi hắn cũng dương hết

Reply:
xong cái âm dương ni là anh merge vô luôn ák

Reply:
ua ben a duoc ma daika